### ISSUE TITLE
Check guidance for markup generators regarding the alt attribute

### ISSUE THREAD

**Issue by agcolom** [w3c/html#885]
On https://www.w3.org/TR/2017/WD-html52-20170406/semantics-embedded-content.html#guidance-for-markup-generators
the 4th paragraph on presence of alt attribute says: "As a last resort, implementors should either set the alt attribute to the empty string, under the assumption that the image is a purely decorative image that doesn’t add any information but is still specific to the surrounding content, or **omit the alt attribute altogether, under the assumption that the image is a key part of the content**." 

I am a bit confused regarding the last part of that sentence (highlighted in bold here). I am not sure why the image being a key part of the content should allow the alt attribute to be unnecessary... 

Any clarification or views would be great. Thank you :-)

**Comment by chaals** [comment]
What happens with `alt=""` is that user technology generally assumes this was done on purpose, because the image doesn't add anything critical, so it's silently ignored. With a missing `alt`, there are various (not very good, as a rule, but sloooowly improving) strategies they can use to try and help the user understand what is going on.

This case is actually covered by [ATAG B 1.1](https://www.w3.org/TR/ATAG/#principle_b1) especially B 1.1.2, the upshot being that creating an `alt` attribute by **guesswork** contravenes that guideline and it should be left out. 

(ATAG 1.0 covered this even more explicitly, but the only difference is you don't have to follow as many links to find out what's required.)

I think here we are likely in conflict with ATAG, and I think they are right and we are wrong.

ping @stevefaulkner @LJWatson Jeanne Spellman

**Comment by agcolom** [comment]
@chaals Thank you very much for the info. That's really useful and I now understand the reason why the alt attribute being missed out completely in this case is better than the ```alt=""```. Maybe we could simply explain that in the spec?

**Comment by scottaohara** [comment]
Hi @chaals and @agcolom.

Was reviewing this issue today and I'd like to take it on, as part of some other editorial work I'd like to do with this page.
